### PR TITLE
Add runtime to create options

### PR DESCRIPTION
**Pull request by ancaantochi**
HostConfig properties used to be serialized/deserialized using Docker.DotNet.Models.HostConfig class which was missing some of the properties that where added to docker since HostConfig class was last update. In order to support all current properties and also future properties, HostConfig class was added to the project with a few properties that are directly used in code and all others were set in the OtherProperties with JsonExtendedData attribute to deserialize properties with no matching class member into the dictionary. 

Having properties in the dictionary is causing issues when using iotedged version 1.0.9 and edgeAgent 1.0.10 and if the property is all lowercase, because iotedged will ignore it. To fix the issue I added all properties that HostConfig had in 1.0.9.

### CODE REVIEW

**Review by darobs — CHANGES_REQUESTED** [review]
I think you're missing 2 fields, but everything else checks out.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/HostConfig.cs [hunk 1]
```diff
@@ -28,6 +28,180 @@ public class HostConfig
...
+        [JsonProperty("ShmSize", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public long ShmSize { get; set; }
+
+        [JsonProperty("GroupAdd", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public IList<string> GroupAdd { get; set; }
+
+        [JsonProperty("InitPath", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string InitPath { get; set; }
+
         [JsonExtensionData]
```
[the remark this answers is not in the document]
Comment (ancaantochi) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/HostConfig.cs:205]: Thanks, ContainerIDFile was missing. DeviceCgroupRules was added later, which version did you compare with? I also updated PidsLimit to nullable type

[the remark this answers is not in the document]
Comment (darobs) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/models/HostConfig.cs:205]: I compared this file to the contents of: https://github.com/Azure/iotedge/blob/release/1.0.9/edgelet/docker-rs/src/models/host_config.rs

I did the comparison by hand, so errors on my part are likely.

**Review by darobs — APPROVED** [review]
:shipit:

**Pull request merged**